### ISSUE TITLE
Feature: Different exit code or ignore Warnings and Notes flag

### ISSUE THREAD

**Issue by cuotos** [koalaman/shellcheck#971]
#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

I want to use shellcheck in an automated build pipeline, but only want it to fail my build if it finds error level problems.

Either a flag to ignore all notes and /or warnings, (--ignore-warnings, --ignore-notes) or maybe different exit codes depending on what was found.

Dan

**Comment by koalaman** [comment]
You could do this today with the help of `jq`:

    shellcheck -f json file | jq -e 'any(.[].level; . == "error")'

It's not currently supported natively because it's less useful than it sounds. There's no clear distinction between `error` and `warning`, it's just a loose sum of accuracy and impact. The plan is to add a verbosity level at some point so that you can choose to get only important warnings/errors.

**Comment by Earnestly** [comment]
Perhaps warnings should not be setting the exit status to non-zero?  This is how virtually every other tool behaves so it's unexpected that shellcheck differs especially when a large portion of the warnings are pedantic.

**Comment by koalaman** [comment]
Which linters and static analysis tools behave like this? `hlint`, `pep8` and `splint` all return non-zero on any trivial finding.

Compilers and interpreters like `gcc` and `javac` aren't really comparable because it's their job to compile or run the code, not to point out potential problems.

**Comment by Earnestly** [comment]
Ah, you may have a point there.  I probably did put too much emphasis on compilers and indeed all the linters and analysers do seem to exit non-zero on anything.

I suppose thinking about it as you've explained, it makes sense for it to behave this way, sorry for the noise

**Comment by cuotos** [comment]
I was wanting to use it as an initial syntax check as part of a pipeline that deployed automation scripts, rather than waiting until it gets run before spotting the issues. 

Im not trying to block badly written scripts, but at least flag non executable ones.

**Comment by kasparsd** [comment]
@cuotos I ended up _grepping_ the json output for `"level":"error"` which seems to work fine:

```shell
shellcheck --format json *.sh | grep --quiet --invert-match '"level":"error"'
```
